lib/ui/platform_dispatcher.dart

```diff
@@ -1981,6 +1981,113 @@ class ViewPadding {
 )
 typedef WindowPadding = ViewPadding;
 
+/// Immutable layout constraints for [FlutterView]s.
+///
+/// Similar to [BoxConstraints], a [Size] respects a [ViewConstraints] if, and
+/// only if, all of the following relations hold:
+///
+/// * [minWidth] <= [Size.width] <= [maxWidth]
+/// * [minHeight] <= [Size.height] <= [maxHeight]
+///
+/// The constraints themselves must satisfy these relations:
+///
+/// * 0.0 <= [minWidth] <= [maxWidth] <= [double.infinity]
+/// * 0.0 <= [minHeight] <= [maxHeight] <= [double.infinity]
+///
+/// For each constraint, [double.infinity] is a legal value.
+///
+/// For a generic class that represents these kind of constraints, see the
+/// [BoxConstraints] class.
+class ViewConstraints {
+  /// Creates view constraints with the given constraints.
+  const ViewConstraints({
+    this.minWidth = 0.0,
+    this.maxWidth = double.infinity,
+    this.minHeight = 0.0,
+    this.maxHeight = double.infinity,
+  });
+
+  /// Creates view constraints that is respected only by the given size.
+  ViewConstraints.tight(Size size)
```

Comment:
`size.width` and `size.height` are invalid constant values, according to Dart (In theory, they could be getters that do some crazy computation using information only available at runtime.). Since they are used in the Initializer list below, the constructor cannot be const, sadly.

(Same applies to the existing `BoxConstraints.tight` constructor)

Reply:
Ahh, right, makes sense. Too bad there's no way to make a final property not be overridable with a getter.